src/machine.ts

```diff
@@ -0,0 +1,122 @@
+import {
+  Machine,
+  assign,
+  send,
+} from 'xstate';
+import { syncToLatestBlock } from './utils';
+import {
+  SyncSchema,
+  SyncContext,
+} from './types';
+
+export const syncHandler = (_context, _event) => (callback, onReceive) => {
+  console.log('Sync handler instantiated');
+  const iterator = syncToLatestBlock();
+  const asyncCall: () => void = async () => {
+    for (;;) {
+      const block = await iterator.next();
+
+      const { value, done } = block;
```

Comment:
What's the types of value and done?

Reply:
Done in 908acaf8781cd29e8ac0735cf83a02f3b6f77cc1

`value` is a `StatusEvent`
`done` is a boolean, indicating if the generator has reached its end

Reply:
Thanks!